process-document/src/main/java/com/ritense/processdocument/service/impl/CamundaProcessJsonSchemaDocumentDeploymentService.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.processdocument.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.ritense.document.domain.event.DocumentDefinitionDeployedEvent;
+import com.ritense.document.domain.impl.Mapper;
+import com.ritense.processdocument.domain.config.ProcessDocumentLinkConfigItem;
+import com.ritense.processdocument.domain.impl.CamundaProcessDefinitionKey;
+import com.ritense.processdocument.domain.impl.request.ProcessDocumentDefinitionRequest;
+import com.ritense.processdocument.service.ProcessDocumentAssociationService;
+import com.ritense.processdocument.service.ProcessDocumentDeploymentService;
+import com.ritense.valtimo.domain.contexts.ContextProcess;
+import com.ritense.valtimo.service.ContextService;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.io.support.ResourcePatternUtils;
+import org.springframework.data.domain.Pageable;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class CamundaProcessJsonSchemaDocumentDeploymentService implements ProcessDocumentDeploymentService {
```

Comment:
This was never covered by tests, right? I do think those need to be added.

Reply:
added